autofs-5.0.7 - fix leaked ldap percent hack allocation in lookup_one()

From: Ian Kent <raven@themaw.net>

Fix a resource leak when calling the percent hack transforms in lookup_one().
---
 modules/lookup_ldap.c |    1 +
 1 file changed, 1 insertion(+)

--- autofs-5.0.5.orig/modules/lookup_ldap.c
+++ autofs-5.0.5/modules/lookup_ldap.c
@@ -2579,6 +2579,7 @@ static int lookup_one(struct autofs_poin
 		if (enc_len1 != 0) {
 			enc_len2 = encode_percent_hack(qKey, &enc_key2, 1);
 			if (enc_len2 < 0) {
+				free(enc_key1);
 				crit(ap->logopt,
 				     "could not use percent hack encode key %s",
 				     qKey);
